pages/pipelines/block_step.md.erb

```diff
@@ -253,6 +253,7 @@ _Optional attributes:_
     <td><code>required</code></td>
     <td>
       A boolean value that defines whether the field is required for form submission.<br>
+      When this value is set to <code>false</code>, non-multiple options will always displayed in a dropdown box, regardless of how many are provided.<br>
       <em>Default:</em> <code>true</code>
     </td>
   </tr>
```

Comment:
Rather than overloading the tables, I think adding the explanation further up the page would be better. You could update the paragraph at line 21 with something like:

> You can add form fields to block steps by adding a `fields` attribute. Block steps with input fields can only be defined using a `pipeline.yml.` There are two field types available: text or select. The select input type displays differently depending on how you configure the options. If you allow people to select multiple options, they display as checkboxes. If you are required to select only one option from six or fewer, they display as radio buttons. Otherwise, the options display in a dropdown menu.

What do you think?

Reply:
@mbelton-buildkite just to clarify your meaning here - do you mean we would add this to line 21 and not add anything on line 256? 

(If so, I'm in two minds about it - 
I get what you mean about not wanting to tables to be too bulked out with text. But, on the other hand,
when I'm looking up something in the docs, if I'm already somewhat familiar with the thing I'm looking up, but wanting to check a detail about it (like in this case), I will usually jump straight to the attributes section as I assume that's where the nitty-gritty mechanics are listed out fully... so if something like this was mentioned above, but not in the table, I might be a bit confused by that.) 

Keen to hear your thoughts here!

Reply:
Yeah, I get why you're torn. What you've described is a common way for experienced users to behave. The question is: Is this information important for experienced users to know when they look up this field? I'm on the fence, so I'm happy to go with your gut and include it in the table. Looking at the rest of the table, it's not out of place.

I do think we should also replace line 21 with what I've got since it describes how the options display. Note, there's the same part on the Input step page to replace.

So, I suggest:

- Updating the text in the table as you're doing.
- Replace line 21 in this file.
- Replace the corresponding text in the Input step file.

Reply:
"Is this information important for experienced users to know when they look up this field?" 
->  this PR came about because a buildkite customer raised an escalation with us because they thought the dropdown menu rendering for 2 options with `required: false` was a bug, since the behaviour wasn't captured in the docs. I can't know the answer for sure, but I think it would have probably helped to have this info in the table in this particular case.

So, I think your final suggestions are good! I'll action those today. Thanks very much for your feedback on this